### ISSUE TITLE
"UnboundLocalError: local variable 'l_value' referenced before assignment"

### ISSUE THREAD

**Issue by stxyzth** [willshersystems/ansible-sshd#32]
I am getting this error when running the default playbook example provided on your README:

```
TASK: [willshersystems.sshd | Configuration] ********************************** 
fatal: [NAP1] => {'msg': "UnboundLocalError: local variable 'l_value' referenced before assignment", 'failed': True}
fatal: [NAP1] => {'msg': "UnboundLocalError: local variable 'l_value' referenced before assignment", 'failed': True}

FATAL: all hosts have already failed -- aborting
```

**Comment by mattwillsher** [comment]
Hi @stxyzth thanks for the report. Which version and Ansible and what OS and architecture?

**Comment by stxyzth** [comment]
ansible 1.9.4
CentOS 6.7 final 64bit

**Comment by mattwillsher** [comment]
Initial guest is that    `` sshd_skip_defaults: true`` is the problem and that the template expects something from there. I'd be interested to hear if setting that false helps.

I'm setting up my own test env. in any case. Not having the example working is a little embarrassing :)

**Comment by mattwillsher** [comment]
I've tried this on fresh install of 6.7 x86_64 with Ansible 1.9.4 and don't get the error. I do get a problem with sandbox being set, so I've adjusted the README, and added a warning in #30 as the example will break SSH access to the server :)

**Comment by stxyzth** [comment]
Any way I can output more information? I am still getting the issue despite doing:

```
    sshd_skip_defaults: false
    sshd_UsePrivilegeSeparation: no
```

**Comment by mattwillsher** [comment]
Add -v to the command line, more v for more verbose up to -vvvv.

It may be worth posting an issue to the Ansible mailing list, or IRC channel, or try a clean install of Ansible. There was a bug in an earlier 1.9 version that looked a bit like this, but that was fixed in 1.9.3.

At this point, I'm not sure whether this is a bug in the role, in Ansible, or something peculiar to your set up.

**Comment by luto** [comment]
I'm hitting the same error with [Ansible fb797a9](https://github.com/ansible/ansible/tree/fb797a9) (pre2.1.0) on CentOS 6.7, Python 2.6.6.

```
TASK [sshd : Configuration] ****************************************************                                                                                                                                                      
task path: roles/sshd/tasks/main.yml:37
<HOST> ESTABLISH SSH CONNECTION FOR USER: None
<HOST> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="~/.ssh/id_rsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=20 -o ControlPath=~/.ansible/cp/ansible-ssh-%h-%p-%r -tt HOST '/bin/sh -c '"'"'( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1453571281.31-222272676732971 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1453571281.31-222272676732971 )" )'"'"''
fatal: [HOST]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"dest": "/etc/ssh/sshd_config", "group": "root", "mode": "0600", "owner": "root", "src": "sshd_config.j2", "validate": "/usr/sbin/sshd -t -f %s"}, "module_name": "template"}, "msg": "UnboundLocalError: local variable 'l_value' referenced before assignment"}
```

**Comment by luto** [comment]
This is caused by [`value` in `sshd_config.j2`](https://github.com/willshersystems/ansible-sshd/blob/master/templates/sshd_config.j2#L26) not having a value assigned at all times. For example the `sshd_Port`-setting isn't defined in the `sshd` or `sshd_defaults` fallback-arrays.

**Comment by mattwillsher** [comment]
That value isn't set would seem to me to be an implicit undefined for value for that block. What's puzzling me is how this has worked until now.

I think rather than set value = undefined I'd rather check if it's defined before calling the render macro. Something about that set strikes me as something that might blow up in the future. I'll get that looked at in the next day or so. (and the change needs to go in meta/10_top.j2, which is used to generate the template)

**Comment by luto** [comment]
I'm also kind of confused by this. But I haven't got the time to look into it properly right now, so I just wanted to get my quick fix out there for other people.

In case you need anything tested, just ping me.

**Comment by mattwillsher** [comment]
I've raised https://github.com/ansible/ansible/issues/14088 to see if the main project can shed any light on the problem. In the meantime, I'll get a fix factored in and cross my fingers all those other undefined sshd_ vars won't break at some point in the future!

**Comment by mattwillsher** [comment]
I've spent some time investigation and implementing a fix. As you've done @luto , set value = undefined needs to be called or else the error is still thrown. It seems that value is seen to be defined, or at least not really undefined in part of the code.  #35 is the first pass at this and now includes a Vagrantfile so I can better test these kind of issues in the future. The new template works for me, verification would be great before I merge.

**Comment by mattwillsher** [comment]
Fixed in PR #35, released up to galaxy as 0.4.2
